rust/lance/src/index/vector.rs

```diff
@@ -173,6 +206,37 @@ pub(crate) async fn build_vector_index(
             pq_params,
         )
         .await?
+    } else if is_ivf_hnsw_pq(stages) {
+        let len = stages.len();
+        let StageParams::Ivf(ivf_params) = &stages[len - 3] else {
+            return Err(Error::Index {
+                message: format!("Build Vector Index: invalid stages: {:?}", stages),
+                location: location!(),
+            });
+        };
+        let StageParams::Hnsw(hnsw_params) = &stages[len - 2] else {
+            return Err(Error::Index {
+                message: format!("Build Vector Index: invalid stages: {:?}", stages),
+                location: location!(),
+            });
+        };
+        let StageParams::PQ(pq_params) = &stages[len - 1] else {
+            return Err(Error::Index {
+                message: format!("Build Vector Index: invalid stages: {:?}", stages),
+                location: location!(),
+            });
+        };
```

Comment:
Since `is_ivf_hnsw_pq` guarantees we will have at least 3 stages I  think we don't need these "else" branches.

Reply:
it's needed for extracting the enum value, just keep it? it may help found that if we changed something like the order of params in `is_ivf_hnsw_pq`